### PR TITLE
Mix in orcid jwt flow with client_credentials flow

### LINKED ISSUE #333: use ORCiD for authn
This ticket consists of two tasks:

1. Create a new API endpoint that calls ORCID's auth url and returns a Token

> Options:
> - the most secure is a two-step process, using an authorization code to request a token
> - alternatively, we could use the implicit flow

2. Add a new authentication flow to the FastAPI OAuth2 model that uses the ORCID access token

- add `openid_oauth2_scheme` here using the authorizationCode flow: https://github.com/microbiomedata/nmdc-runtime/blob/ebe41c9874ccf0326b925462afd6f8f672b6671e/nmdc_runtime/api/core/auth.py#L116
(alt: we could also use the implicit openid flow, as described in the orcid dev tools)

DOCS: [fastapi.security.OAuthFlows.authorizationCode](https://fastapi.tiangolo.com/reference/openapi/models/#fastapi.openapi.models.OAuthFlows.authorizationCode)

### PR DESCRIPTION
**Pull request by dwinston**
User can load `/orcid_authorize` to obtain orcid JWT. Then authorize using `client_credentials` flow using orcid jwt as `client_id` and empty `client_secret`. Will create user with orcid as username if it doesn't exist. Returns a user token just like the username/password flow.

closes #333

### CODE REVIEW

File: nmdc_runtime/api/core/auth.py [hunk 1]
```diff
@@ -99,17 +111,21 @@ async def __call__(self, request: Request) -> Optional[str]:
         scheme, param = get_authorization_scheme_param(authorization)
         if not authorization or scheme.lower() != "bearer":
             if self.auto_error:
+                print("uh oh")
```
Comment (eecavanna) [inline nmdc_runtime/api/core/auth.py:114]: Marking this as something someone may want to revisit (i.e. elaborate on, remove) as part of the upcoming "simplify auth" work.

File: nmdc_runtime/api/endpoints/users.py [hunk 2]
```diff
@@ -22,6 +32,40 @@
...
+    """NOTE: You want to load /orcid_authorize directly in your web browser to initiate the login redirect flow."""
+    return RedirectResponse(
+        f"https://orcid.org/oauth/authorize?client_id={ORCID_CLIENT_ID}"
+        "&response_type=token&scope=openid&"
+        f"redirect_uri={BASE_URL_EXTERNAL}/orcid_token"
+    )
+
+
+@router.get("/orcid_token")
+async def redirect_uri_for_orcid_token(req: Request):
```
Comment (eecavanna) [inline nmdc_runtime/api/endpoints/users.py:46]: Add docstring:

```py
r"""
Returns the HTML of a web page that...
"""
```

File: nmdc_runtime/api/core/auth.py [hunk 3]
```diff
@@ -16,6 +16,18 @@
 
 SECRET_KEY = os.getenv("JWT_SECRET_KEY")
 ALGORITHM = "HS256"
+ORCID_CLIENT_ID = os.getenv("ORCID_CLIENT_ID")
+
+# https://orcid.org/.well-known/openid-configuration
+# XXX do we want to live-load this?
```
Comment (eecavanna) [inline nmdc_runtime/api/core/auth.py:22]: If the JSON content at https://orcid.org/oauth/jwks is something that changes over time (I don't know whether it is), I'd prefer to live-load it.

**Review by eecavanna — DISMISSED** [review]
I left a few comments about documentation and print statements.

**Comment by dwinston** [comment]
@eecavanna great feedback, thanks.

File: nmdc_runtime/api/core/auth.py (hunk 3 above)
Comment (dwinston) [reply nmdc_runtime/api/core/auth.py:22]: hmm. yeah, will include in refactoring.

**Pull request merged**